app/iot/service.py

```diff
@@ -12,38 +13,37 @@ def generate_id(length: int = 8) -> str:
 # Protocol is very similar to ABC, but uses duck typing
 # so devices should not inherit for it (if it walks like a duck, and quacks like a duck, it's a duck)
 class Device(Protocol):
-    def connect(self) -> None:
+    async def connect(self) -> None:
         ...  # Ellipsis - similar to "pass", but sometimes has different meaning
 
-    def disconnect(self) -> None:
+    async def disconnect(self) -> None:
         ...
 
-    def send_message(self, message_type: MessageType, data: str) -> None:
+    async def send_message(self, message_type: MessageType, data: str) -> None:
         ...
 
 
 class IOTService:
     def __init__(self):
         self.devices: dict[str, Device] = {}
 
-    def register_device(self, device: Device) -> str:
-        device.connect()
+    async def register_device(self, device: Device) -> str:
+        await device.connect()
         device_id = generate_id()
         self.devices[device_id] = device
         return device_id
 
-    def unregister_device(self, device_id: str) -> None:
-        self.devices[device_id].disconnect()
+    async def unregister_device(self, device_id: str) -> None:
+        await self.devices[device_id].disconnect()
         del self.devices[device_id]
 
     def get_device(self, device_id: str) -> Device:
         return self.devices[device_id]
 
-    def run_program(self, program: list[Message]) -> None:
-        print("=====RUNNING PROGRAM======")
-        for msg in program:
-            self.send_msg(msg)
-        print("=====END OF PROGRAM======")
+    # async def run_program(self, program: list[Message]) -> None:
+    #     print("=====RUNNING PROGRAM======")
+    #     await asyncio.gather(*[self.send_msg(msg) for msg in program])
+    #     print("=====END OF PROGRAM======")
```

Comment:
> These comments are redundant, I see that you don't use this method

@Serhii-Leonenko 
I left it in case I need these prints for this task, but if everything is fine and they are optional, I will remove this part of the code